### PR TITLE
feat: shortcode for openapi

### PR DESCRIPTION
**Pull request by Isammoc**
Motivation:
Publish the API documentation

Modifications:
 * openapi shortcode
 * custom scss (to avoid pre background)

Result:
Can use openapi

### CODE REVIEW

File: layouts/shortcodes/swaggerui.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+<script>
+  window.onload = function () {
+    const ui = SwaggerUIBundle({
+      url: {{ $original | relURL }},
+    dom_id: '#swagger_ui',
+      presets: [
+      SwaggerUIBundle.presets.apis,
+      SwaggerUIStandalonePreset
+    ]
+  });
```
Comment (notdryft) [inline layouts/shortcodes/swaggerui.html:12]:
```suggestion
  window.onload = function () {
    const ui = SwaggerUIBundle({
      url: {{ $original | relURL }},
      dom_id: '#swagger_ui',
      presets: [
        SwaggerUIBundle.presets.apis,
        SwaggerUIStandalonePreset
      ]
    });
```

**Review by notdryft — CHANGES_REQUESTED** [review]

File: layouts/partials/footer/script-footer.html [hunk 2]
```diff
@@ -52,3 +52,5 @@
   */}}
   <script src="{{ $index.Permalink }}" integrity="{{ $index.Data.Integrity }}" crossorigin="anonymous"></script>
 {{ end -}}
+<script src="https://unpkg.com/swagger-ui-dist@5.9.0/swagger-ui-bundle.js" crossorigin="anonymous"></script>
+<script src="https://unpkg.com/swagger-ui-dist@5.9.0/swagger-ui-standalone-preset.js" crossorigin="anonymous"></script>
```
Comment (tpetillot) [inline layouts/partials/footer/script-footer.html:56]: Swagger UI is heavy (https://bundlephobia.com/package/swagger-ui@5.9.0), it shouldn't be downloaded from everywhere in the documentation except pages using the shortcode...

Comment (Isammoc) [reply layouts/partials/footer/script-footer.html:56]: Yep, I know...
This is a temporary implementation.
I don't want to let an external website access (unpkg.com)

**Comment by Isammoc** [comment]
Thanks to @notdryft, we found that "`swagger-ui`" npm module does not work, but "`swagger-ui-dist`" does.

So I can make it works and better. only call it once at the end if needed thanks to the `.Page.Store.Set` and `.Page.Store.Get`

File: layouts/partials/head/stylesheet.html [hunk 3]
```diff
@@ -9,3 +9,15 @@
   <link rel="stylesheet" href="{{ $secureCSS.Permalink }}" integrity="{{ $secureCSS.Data.Integrity }}" crossorigin="anonymous">
 {{ end -}}
 <noscript><style>img.lazyload { display: none; }</style></noscript>
+{{ if .Page.Store.Get "hasOpenApi" -}}
```
Comment (notdryft) [inline layouts/partials/head/stylesheet.html:12]: Store is persistent between builds but it is only tied to a single page at a time? Does it work on first build? oO

**Review by notdryft — CHANGES_REQUESTED** [review]

**Review by tpetillot — APPROVED** [review]
Approved in condition of @notdryft comment

File: layouts/partials/head/stylesheet.html (hunk 3 above)
Comment (Isammoc) [reply layouts/partials/head/stylesheet.html]: Store and Scratch work the same for build.
The difference is only when using hugo as "server"

**Review by notdryft — APPROVED** [review]

**Pull request merged**